src/components/TodoItem.tsx

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useCallback } from 'react';
+import { Todo } from '../types/Todo';
+import classNames from 'classnames';
+import { deleteTodos, editTodos } from '../api/todos';
+import { useTodosContext } from './useTodosContext';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const {
+    list,
+    setList,
+    editingTodo,
+    setEditingTodo,
+    handleError,
+    setLoadingTodosIds,
+    setErrorMessage,
+    loadingTodosIds,
+  } = useTodosContext();
+
+  const handleCheckbox = (checkedTodo: Todo) => {
+    const updatedList = list.map((item: Todo) => {
+      if (item.id === checkedTodo.id) {
+        return { ...item, completed: !item.completed };
+      }
+
+      return item;
+    });
+
+    setList(updatedList);
+  };
+
+  const handleTodoDelete = useCallback(
```

Comment:
You don't need wrap with `useCallback` if you don't pass your function as in the child component

Reply:
have this error without callback

<img width="476" alt="image" src="https://github.com/mate-academy/react_todo-app-loading-todos/assets/113982603/5492e48b-c0e0-41b1-b5c5-72a8b8380901">

Reply:
> have this error without callback
> 
> <img alt="image" width="476" src="https://private-user-images.githubusercontent.com/113982603/318304402-5492e48b-c0e0-41b1-b5c5-72a8b8380901.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE5NTI3NjcsIm5iZiI6MTcxMTk1MjQ2NywicGF0aCI6Ii8xMTM5ODI2MDMvMzE4MzA0NDAyLTU0OTJlNDhiLWMwZTAtNDFiMS1iNWM1LTcyYThiODM4MDkwMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwMVQwNjIxMDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03M2EyOTU4ZDRjYTdkMjY1ZDljODBiMDA4OTVhY2Q4MWFlNmNiMjRmMDNiYjkzYzI2YjkwOGI2NDI5MTFlOWE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.v8WYV_oLsUzxpNKIj14gydVMHybEUveG72ndIV9w8hc">

I don't see what you have at line 77, but it doesn't make sense to wrap all with `useCallback`

Reply:
its used in the dependency array of another function